direct/nn/mri_models.py

```diff
@@ -423,6 +600,18 @@ def grad_l2_loss(
                 loss_dict[loss_fn] = multiply_function(curr_loss.multiplier, nrmse_loss)
             elif loss_fn in ["nmae_loss", "kspace_nmae_loss"]:
                 loss_dict[loss_fn] = multiply_function(curr_loss.multiplier, nmae_loss)
+            elif loss_fn in ["snr_loss", "psnr_loss"]:
+                loss_dict[loss_fn] = multiply_function(
+                    curr_loss.multiplier, (snr_loss if loss_fn == "snr_loss" else psnr_loss)
+                )
+            elif loss_fn == "hfen_l1_loss":
+                loss_dict[loss_fn] = multiply_function(curr_loss.multiplier, hfen_l1_loss)
+            elif loss_fn == "hfen_l2_loss":
+                loss_dict[loss_fn] = multiply_function(curr_loss.multiplier, hfen_l2_loss)
+            elif loss_fn == "hfen_l1_norm_loss":
+                loss_dict[loss_fn] = multiply_function(curr_loss.multiplier, hfen_l1_norm_loss)
+            elif loss_fn == "hfen_l2_norm_loss":
+                loss_dict[loss_fn] = multiply_function(curr_loss.multiplier, hfen_l2_norm_loss)
```

Comment:
Perhaps it's better tp turn this into a NamedTuple, or some sort, subtyping it also from string. Then you can loss_fn == that string, but also do loss_fn calling directly as a function.

Reply:
Done!